src/utils/profile.ts

```diff
@@ -0,0 +1,17 @@
+import firebase from "firebase/app";
```

Comment:
Do we want to create a new 'prefix' for firestore stuff? eg. `api/` (as opposed to `utils/`)? Would help to keep things cleaner/isolated/easier to find all of the firebase calls/etc. And then can easily access them from thunks/etc as required in our redux/etc layer.

Reply:
Yes- eventually. Another opportunity along the same lines is pulling `firestore` references out of our selectors across the codebase. A couple key things keep us moving in this direction, which we can keep eyes on in future reviews and make sure we progressively hew to:

* Ensure the selectors we encounter come from `selectors.ts` or get converted to use those selectors as part of the PR, so we can one day make them point somewhere other than `store.firestore`
* Move firestore queries progressively into hooks (combining `useFirestoreConnect` and `useSelector` into one call ala `const venue = useCurrentVenue()`, and then into the store (using thunks if needed)

These types of directions we set offer a low-cost, incremental, PR and code-review oriented way to push Firebase down into its own layer. Any other ideas are welcome!

Reply:
We can probably eslint warn/ban imports of the 'direct' firebase import references, + the firebase hooks, etc outside of our chosen 'allowed locations' to aid in this.

Reply:
^tracked as https://github.com/sparkletown/internal-sparkle-issues/issues/2

Reply:
Yes; banning hooks/* and firebase imports in molecules will be a good start.

Reply:
Not sure banning `hooks/*` in molecules is correct.. is it? That would mean none of our custom hooks?

Reply:
The underlying goal is to prevent store access in the “dumb” components so they can receive data only via props and are closer to being controlled components. So maybe we would separate out the hooks which access the store from the rest, or achieve that goal a different way. Definitely a discussion for the future.

Reply:
Ah, makes sense. I need to find time to actually read that intro post/primer on the whole organism/etc pattern we're using at some point.... :p

Reply:
> I need to find time to actually read that intro post/primer on the whole organism/etc pattern we're using at some point.... 😝

So do I! 😂 I inferred from context in the end. We inherited it from an agency a while back and I suspect it’s meant to encourage the controlled/uncontrolled component divide.

Reply:
Haha, that makes sense. I have it bookmarked.. 😛